src/rpc/blockchain.cpp

```diff
@@ -545,31 +545,50 @@ UniValue backupprivatekeys(const UniValue& params, bool fHelp)
 
 UniValue rainbymagnitude(const UniValue& params, bool fHelp)
     {
-    if (fHelp || (params.size() < 2 || params.size() > 3))
+    if (fHelp || (params.size() < 2 || params.size() > 4))
         throw runtime_error(
-                "rainbymagnitude <whitelisted project> <amount> [message]\n"
+                "rainbymagnitude <whitelisted project> <amount> [trial run] [output details]\n"
                 "\n"
-                "<whitelisted project> --> Required: If a project is specified, rain will be limited to that project. Use * for network-wide.\n"
-                "<amount> --> Required: Specify amount of coins to be rained in double precision float\n"
-                "[message] -> Optional: Provide a message rained to all rainees\n"
+                "<whitelisted project> -> Required: If a project is specified, rain will be limited to that project.\n"
+                "                         Use * for network-wide.\n"
+                "<amount>              -> Required: Specify amount of coins to be rained in decimal (1000 GRC minimum).\n"
+                "                         This will be rounded to the nearest Halford (0.00000001)\n"
+                "[trial run]           -> Optional: Boolean to specify a trial run instead of actual transaction\n"
+                "                         (default false)\n"
+                "[output details]      -> Optional: Boolean to output details of rain (default false if not\n"
+                "                         trial run, true if trial run).\n"
```

Comment:
Explanation:

 - `project_id`: clarify that this must be an exact project ID from a contract
 - `amount`: it's parsed like all other amounts now, no need for special annotation
 - `trial_run`: minor formatting 
 - `output_details`: minor formatting

```suggestion
                "project_id     -> Required: Limits rain to a specific project. Use \"*\" for\n"
                "                  network-wide. Call \"listprojects\" for the IDs of eligible\n"
                "                  projects.
                "amount         -> Required: Amount to rain (1000 GRC minimum).\n"
                "trial_run      -> Optional: Boolean to specify a trial run instead of an actual\n"
                "                  transaction (default: false).\n"
                "output_details -> Optional: Boolean to output recipient details (default: false\n"
                "                  if not trial run, true if trial run).\n"
```